sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteTables.java

```diff
@@ -251,7 +253,10 @@ public void processElement(
       List<String> partitionFiles = Lists.newArrayList(element.getValue().getFilenames());
       String jobIdPrefix =
           BigQueryResourceNaming.createJobIdWithDestination(
-              c.sideInput(loadJobIdPrefixView), tableDestination, partition, c.pane().getIndex());
+              c.sideInput(loadJobIdPrefixView),
+              tableDestination,
+              partition,
+              element.getValue().getPaneIndex());
```

Comment:
My concern is that if we have this:

    GBK -> processing -> reshuffle -> this DoFn

Can you tell me which case we are in?

1. The `processing` left the key the same so the reshuffle is just for checkpoint
2. The `processing` changed the key so that the current element key + pane index is no longer unique

Reply:
Well it's a little bit more subtle. It is the same key, but became ShardedKey after `processing`

After GBK, key is DestinationT
After processing, key is ShardedKey<DestinationT> (shard is added in WritePartitions)